consensus/spos/consensusMessageValidator.go

```diff
@@ -0,0 +1,270 @@
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Header) = %d SignatureShare = %s PubKeysBitmap = %s AggregateSignature = %s LeaderSignature = %s",
+			ErrInvalidMessage,
+			len(cnsMsg.Header),
+			logger.DisplayByteSlice(cnsMsg.SignatureShare),
+			logger.DisplayByteSlice(cnsMsg.PubKeysBitmap),
+			logger.DisplayByteSlice(cnsMsg.AggregateSignature),
+			logger.DisplayByteSlice(cnsMsg.LeaderSignature))
+	}
+
+	if len(cnsMsg.Body) > core.MegabyteSize {
+		return fmt.Errorf("%w : received body from consensus topic has an invalid size: %d",
+			ErrInvalidBodySize,
+			len(cnsMsg.Body))
+	}
+
+	return nil
+}
+
+func (wrk *Worker) checkMessageWithBlockHeaderValidity(cnsMsg *consensus.Message) error {
+	isMessageInvalid := cnsMsg.Body != nil ||
+		cnsMsg.SignatureShare != nil ||
+		cnsMsg.PubKeysBitmap != nil ||
+		cnsMsg.AggregateSignature != nil ||
+		cnsMsg.LeaderSignature != nil
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Body) = %d SignatureShare = %s PubKeysBitmap = %s AggregateSignature = %s LeaderSignature = %s",
+			ErrInvalidMessage,
+			len(cnsMsg.Body),
+			logger.DisplayByteSlice(cnsMsg.SignatureShare),
+			logger.DisplayByteSlice(cnsMsg.PubKeysBitmap),
+			logger.DisplayByteSlice(cnsMsg.AggregateSignature),
+			logger.DisplayByteSlice(cnsMsg.LeaderSignature))
+	}
+
+	if cnsMsg.Header == nil || len(cnsMsg.Header) > core.MegabyteSize {
+		return fmt.Errorf("%w : received header from consensus topic has an invalid size: %d",
+			ErrInvalidHeaderSize,
+			len(cnsMsg.Header))
+	}
+
+	return nil
+}
+
+func (wrk *Worker) checkMessageWithSignatureValidity(cnsMsg *consensus.Message) error {
+	isMessageInvalid := cnsMsg.Body != nil ||
+		cnsMsg.Header != nil ||
+		cnsMsg.PubKeysBitmap != nil ||
+		cnsMsg.AggregateSignature != nil ||
+		cnsMsg.LeaderSignature != nil
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Body) = %d len(Header) = %d PubKeysBitmap = %s AggregateSignature = %s LeaderSignature = %s",
+			ErrInvalidMessage,
+			len(cnsMsg.Body),
+			len(cnsMsg.Header),
+			logger.DisplayByteSlice(cnsMsg.PubKeysBitmap),
+			logger.DisplayByteSlice(cnsMsg.AggregateSignature),
+			logger.DisplayByteSlice(cnsMsg.LeaderSignature))
+	}
+
+	if len(cnsMsg.SignatureShare) != core.SignatureSizeInBytes {
+		return fmt.Errorf("%w : received signature share from consensus topic has an invalid size: %d",
+			ErrInvalidSignatureSize,
+			len(cnsMsg.SignatureShare))
+	}
+
+	return nil
+}
+
+func (wrk *Worker) checkMessageWithFinalInfoValidity(cnsMsg *consensus.Message) error {
+	isMessageInvalid := cnsMsg.Body != nil ||
+		cnsMsg.Header != nil ||
+		cnsMsg.SignatureShare != nil
+
+	if isMessageInvalid {
+		return fmt.Errorf("%w : received message from consensus topic is invalid: "+
+			"len(Body) = %d len(Header) = %d SignatureShare = %s",
```

Comment:
is this error message not too long ? maybe do a log trace before then send and simpler wrapped error.

Reply:
Done